gauge_api_steps/reporting.py

```diff
@@ -42,10 +43,26 @@ def report_response_info(resp: HTTPResponse|HTTPError, resp_body: bytes) -> None
 
 
 def print_and_report(message: str) -> None:
+    masked_message = _mask_secrets(message)
     replace_whitespace = os.environ.get("replace_whitespace_in_console")
-    console_message = message
+    console_message = masked_message
     if replace_whitespace is not None:
         console_message = console_message.replace(' ', replace_whitespace)
         console_message = console_message.replace('\t', replace_whitespace * 4)
     print(console_message)
-    Messages.write_message(message.replace('&', '&amp;').replace('<', '&lt;').replace('>', '&gt;').replace('\t', '    ').replace(' ', '&nbsp;'))
+    Messages.write_message(masked_message.replace('&', '&amp;').replace('<', '&lt;').replace('>', '&gt;').replace('\t', '    ').replace(' ', '&nbsp;'))
+
+
+def _mask_secrets(message: str) -> str:
+    secret_props_prop = os.environ.get("mask_secrets")
```

Comment:
IMHO the name of the variable is a bit strange. Why not "secret_prop" or "secrect_property"?

Reply:
That's because it is not the secret property itself, but it is a property whose value references the actual secret properties.